src/isql/extract.epp

```diff
@@ -3594,3 +3568,172 @@ static void list_views()
 		return;
 	END_ERROR;
 }
+
+static void list_external_procedures(SSHORT default_char_set_id)
+{
+/**************************************
+ *
+ *	l i s t _ e x t e r n a l _ p r o c e d u r e s
+ *
+ **************************************
+ *
+ * Functional description
+ * Create all external procedures declarations.
+ *
+ **************************************/
+
+	fb_assert(isqlGlob.major_ods >= ODS_VERSION12);
```

Comment:
Current ISQL (FB 3-4-5) cannot extract procedures/functions from pre-ODS12 databases, so nothing really changed.

Reply:
Both `list_external_procedures()` and `list_external_functions()` is called unconditionally by `EXTRACT_ddl()`, see

https://github.com/FirebirdSQL/firebird/pull/8054/files#diff-580a2883156750bdebc94df7a24f090f8b288b49558292ae66fde9b21cbbc432R184

so, what happens in case of pre-ODS13 DB ? 
Error will be raised ? 
Nothing, as query below returns no rows ?
Why assert then ? It will not make DEBUG build happy, AFAIU.
I just want it to be clear.

Reply:
Error "field does not exist". You may find the same assert in `list_procedure_bodies()` which is also executed unconditionally. AFAIU, the only purpose of the assert is to warn that the extract function is not able to deal with older ODS. We may remove these asserts separately, if you disagree with their usage in general. They are not related to this PR and existed before.

Reply:
Strange logic, as for me - use assert and runtime error instead of condition and correct handling of old ODS.
I agree, it is not related with this PR. I never looked into this code before and thus wondering.

Reply:
IIRC, it was discussed in fb-devel and we agreed to simplify the code this way, but it was ages ago...